cli/cmd/edges.go

```diff
@@ -0,0 +1,325 @@
+	return &edgesOptions{
+		namespace:    "",
+		outputFormat: tableOutput,
+	}
+}
+
+type indexedEdgeResults struct {
+	ix   int
+	rows []*pb.Edge
+	err  error
+}
+
+func newCmdEdges() *cobra.Command {
+	options := newEdgesOptions()
+
+	cmd := &cobra.Command{
+		Use:   "edges [flags] (RESOURCETYPE)",
+		Short: "Display connections between resources, and Linkerd proxy identities",
+		Long: `Display connections between resources, and Linkerd proxy identities.
+
+  The RESOURCETYPE argument specifies the type of resource to display edges within. A namespace must be specified.
+
+  Examples:
+  * deploy
+  * ds
+  * job
+  * po
+  * rc
+  * sts
+
+  Valid resource types include:
+  * daemonsets
+  * deployments
+  * jobs
+  * pods
+  * replicationcontrollers
+  * statefulsets`,
+		Example: `  # Get all edges between pods in the test namespace.
+  linkerd edges po -n test`,
+		Args:      cobra.ExactArgs(1),
+		ValidArgs: util.ValidTargets,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			reqs, err := buildEdgesRequests(args, options)
+			if err != nil {
+				return fmt.Errorf("error creating edges request: %s", err)
+			}
+
+			// The gRPC client is concurrency-safe, so we can reuse it in all the following goroutines
+			// https://github.com/grpc/grpc-go/issues/682
+			client := checkPublicAPIClientOrExit()
+			c := make(chan indexedEdgeResults, len(reqs))
+			for num, req := range reqs {
+				go func(num int, req *pb.EdgesRequest) {
+					resp, err := requestEdgesFromAPI(client, req)
+					rows := edgesRespToRows(resp)
+					c <- indexedEdgeResults{num, rows, err}
+				}(num, req)
+			}
+
+			totalRows := make([]*pb.Edge, 0)
+			i := 0
+			for res := range c {
+				if res.err != nil {
+					return res.err
+				}
+				totalRows = append(totalRows, res.rows...)
+				if i++; i == len(reqs) {
+					close(c)
+				}
+			}
+
+			output := renderEdgeStats(totalRows, options)
+			_, err = fmt.Print(output)
+
+			return err
+		},
+	}
+
+	cmd.PersistentFlags().StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace of the specified resource")
+	cmd.PersistentFlags().StringVarP(&options.outputFormat, "output", "o", options.outputFormat, "Output format; one of: \"table\" or \"json\"")
+	return cmd
+}
+
+func validateEdgesOutputFormat(options *edgesOptions) error {
+	switch options.outputFormat {
+	case tableOutput, jsonOutput:
+		return nil
+	default:
+		return fmt.Errorf("--output currently only supports %s and %s", tableOutput, jsonOutput)
+	}
+}
+
+func buildEdgesRequests(resources []string, options *edgesOptions) ([]*pb.EdgesRequest, error) {
+	targets, err := util.BuildResources(options.namespace, resources)
+
+	if err != nil {
+		return nil, err
+	}
+	err = validateEdgesOutputFormat(options)
+	if err != nil {
+		return nil, err
+	}
+
+	requests := make([]*pb.EdgesRequest, 0)
+	for _, target := range targets {
+		if target.Name != "" {
+			target.Name = "" // validating that no resource name is specified in the request
```

Comment:
I think it'd be nicer to make a validation error here to let the user know that they can't get edges for a specific resource name, it might be confusing to drop this silently and return results anyway